app/apollo/resolvers/cluster.js

```diff
@@ -577,19 +652,20 @@ const clusterResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, registration }, `${queryName} validating`);
-
-        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context);
-
-        logger.info({ req_id, user, org_id, registration }, `${queryName} validating - authorized`);
+        logger.info({req_id, user, org_id, registration}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateJson( 'registration', registration );
+
+        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context, [registration.name]);
```

Comment:
valid auth needs to move to after simple checks like verifying that `registration.name` exists and is valid.  No need to check auth if the input is bad.

Reply:
Still a problem, wasn't addressed.

Reply:
Fixed, I decided to leave it previously based on context of validateName and registration.name

Reply:
Still a problem, not addressed.

Aside from being sub-optimal performance-wise, leaving the `validAuth` call where it is located instead of moving it. after the simple checks could cause an exception to be thrown.  If the `registration` input object does not have `registration.name` attribute at all, the validAuth call will likely error because `registration.name` is passed to it.  It shouldn't bother checking auth till after it has done the `if (!registration.name) {` check, which is currently _after_ this auth line.

The `validating - authorized` log point is also out of order as a result -- it should be immediately after the auth check, but is instead immediately after the `if (!registration.name) {` check which is confusing.

Reply:
I moved validAuth() below if (!registration.name) and moved the log message below validAuth()

Reply:
Resolved, but added new comment about changes to original validations